versioned_docs/version-4.0.0-alpha.1/datastructures/storage-in-metadata.md

```diff
@@ -3,6 +3,209 @@ title: Metadata Format
 slug: /datastructures/storage-in-metadata
 ---
 
-:::caution
-TODO: Add documentation on the `storage ` field in the metadata.
-:::
+The storage layout of a contract is reflected inside the metadata. It allows third-party 
+tooling to work with contract storage and can also help to better understand the storage 
+layout of any given contract.
+
+Given a contract with the following storage:
+
+```rust
+#[ink(storage)]
+pub struct MyContract {
+    balance: Balance,
+    block: BlockNumber,
+    something: Lazy<bool>,
+}
+```
+
+The storage will be reflected inside the metadata as like follows:
+
+```json
+"root": {
+  "layout": {
+    "struct": {
+      "fields": [
+        {
+          "layout": {
+            "leaf": {
+              "key": "0x00000000",
+              "ty": 0
+            }
+          },
+          "name": "balance"
+        },
+        {
+          "layout": {
+            "leaf": {
+              "key": "0x00000000",
+              "ty": 1
+            }
+          },
+          "name": "block"
+        },
+        {
+          "layout": {
+            "root": {
+              "layout": {
+                "leaf": {
+                  "key": "0xb1f4904e",
+                  "ty": 2
+                }
```

Comment:
Not sure in how much detail we want to talk about this in the docs, but there are more keys than just `root` and `leaf` available to describe items in storage. E.g we can have a root key that contains an `array` or `enum` layout

Reply:
Hmm yeah, I think it is a good point but I failed to bring up an example covering everything that is also small and easy to digest...